Tests/PayPalMessagesTests/PayPalMessageModalTests.swift

```diff
@@ -0,0 +1,176 @@
+import Foundation
+import XCTest
+@testable import PayPalMessages
+
+class PayPalMessageModalTests: XCTestCase {
+    let config = PayPalMessageModalConfig(
+        data: .init(
+            clientID: "Test123",
+            environment: .sandbox,
+            amount: 100.0,
+            currency: "USD",
+            offerType: .payLaterLongTerm
+        )
+    )
+
+    var modalViewController: PayPalMessageModal!
+
+    override func setUpWithError() throws {
+        // Create an instance of PayPalMessageModal for testing
+        modalViewController = PayPalMessageModal(config: config)
+        modalViewController.loadViewIfNeeded()
+    }
+
+    override func tearDownWithError() throws {
+        // Clean up after each test
+        modalViewController = nil
+    }
+
+    func testShowWithConfig() {
+        let presentingViewController = UIViewController()
+
+        // Getting the all scenes
+        let scenes = UIApplication.shared.connectedScenes
+        // Getting windowScene from scenes
+        let windowScene = scenes.first as? UIWindowScene
+        // Getting window from windowScene
+        let window = windowScene?.windows.first
+        window?.rootViewController = presentingViewController
+
+        presentingViewController.loadViewIfNeeded()
+        modalViewController.loadViewIfNeeded()
+
+        // Create an expectation to track the modal presentation
+        let presentationExpectation = XCTestExpectation(description: "Modal presentation")
+
+        modalViewController.show(config: config) // Replace with your mock configuration
+
+        // Add a brief delay for animations to complete
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { // Increase the delay if needed
+            // Verify that the modal view controller is presented modally
+            XCTAssertTrue(self.modalViewController.modalPresentationStyle == .formSheet)
+
+            // Fulfill the presentation expectation
+            presentationExpectation.fulfill()
+        }
+
+        // Wait for the presentation expectation with an appropriate timeout
+        wait(for: [presentationExpectation], timeout: 5.0)
+    }
+
+
+    func testSupportedInterfaceOrientations() {
+        let orientations = modalViewController.supportedInterfaceOrientations
+
+        XCTAssertEqual(orientations, .portrait)
+    }
+
+    func testPreferredInterfaceOrientationForPresentation() {
+        let orientation = modalViewController.preferredInterfaceOrientationForPresentation
+
+        XCTAssertEqual(orientation, .portrait)
+    }
+
+    func testShouldAutorotate() {
+        let shouldAutoRotate = modalViewController.shouldAutorotate
+
+        XCTAssertFalse(shouldAutoRotate)
+    }
```

Comment:
Would it reduce some noise if these were just combined into a single test that just tests initial property values?